logging/logging.go

```diff
@@ -1,25 +1,55 @@
 package logging
 
 import (
+	"github.com/Jaggernaut555/respecbot-v2/types"
 	"log"
 	"os"
+	"strings"
+	"time"
 )
 
 var (
 	logger    *log.Logger
 	errLogger *log.Logger
+
+	// An API instance to send log messages to
+	apiInstance types.API
 )
 
 func init() {
 	logger = log.New(os.Stdout, "", log.Ldate|log.Ltime)
 	errLogger = log.New(os.Stderr, "ERR - ", log.Ldate|log.Ltime)
 }
 
+func SetAPIInstance(instance types.API) {
+	apiInstance = instance
+}
+
 //Log Log all the given data
 func Log(data ...string) {
 	logger.Print(data)
 }
 
+//Log that logs given data and sends to specific server channels
+func LogToServer(server *types.Server, data ...string) {
```

Comment:
We should have logging instead register a callback function of the respective server's API type, rather than storing a reference to the API itself here. Send the parameters to the correct API type, and then have the API determine where it should be logging in the callback.